zapcore/sampler.go

```diff
@@ -29,6 +30,14 @@ import (
 const (
 	_numLevels        = _maxLevel - _minLevel + 1
 	_countersPerLevel = 4096
+
+	// ReportingLoggerName is the default name of logger that generates sampling
```

Comment:
can we avoid exporting these (as they become part of our API that we have to support forever, which means we can't move them into a subpackage in future, etc)

instead, let's document it in the sampler config?